yt/yt/server/exec/program.h

```diff
@@ -154,12 +170,14 @@ class TExecProgram
         args.push_back(nullptr);
 
         // We are ready to execute user code, send signal to JobProxy.
-        try {
-            auto jobProxyControl = CreateUserJobSynchronizerClient(config->UserJobSynchronizerConnectionConfig);
-            jobProxyControl->NotifyExecutorPrepared();
-        } catch (const std::exception& ex) {
-            LogToStderr(Format("Unable to notify job proxy\n%v", ex.what()));
-            Exit(5);
+        if (config->UserJobSynchronizerConnectionConfig) {
```

Comment:
Let's add a comment that config is absent for job shell.

Also, `if (const auto& config = ...)` can be used